addons/account_invoice_import/wizard/account_invoice_send.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+import base64
+
+
+class AccountInvoiceSend(models.TransientModel):
+    _inherit = 'account.invoice.send'
+    _description = 'Account Invoice Send'
+
+    is_xml = fields.Boolean('XML')
+
+    def send_and_print_action(self):
+        self.ensure_one()
+        if self.is_xml:
+            for invoice in self.invoice_ids:
+                to_attach = [xml_export() for xml_type, xml_checker, xml_import, xml_export, embed_to_pdf in invoice._get_xml_decoders() if not embed_to_pdf]
+
+                for filename, content in to_attach:
+                    attachment_id = self.env['ir.attachment'].search([('name', '=', filename)], limit=1)
+                    if attachment_id:
+                        attachment_id.update({'datas': base64.encodebytes(content)})
```

Comment:
If I well understand, my ubl file is updated when sending the invoice? This is weird...